### ISSUE TITLE
Fail to run `protolint lint .` and get go syntax error

### ISSUE THREAD

**Issue by DoneSpeak** [yoheimuta/protolint#218]
I install protolint with `brew install protolint` and run `protolint lint .` to verify the following proto file.
```proto
message Book {
    optional string name = 1;
}
```
but getting error result:
```bash
$ protolint lint .                                                       
found "\"message\"(Token=27, Pos=tests/test.proto:1:1)" but expected [syntax] at /home/runner/go/pkg/mod/github.com/yoheimuta/go-protoparser/v4@v4.5.3-0.20220212052745-6d78fe6fbe7b/parser/syntax.go:62. Use -v for more details
```

How can I fix the problem?

What's more, I want to create my Dockerfile installed protolint and using the image to run pre-commit. I get a same result in the ci.

The project: https://gitlab.com/DoneSpeak/container-image

Here is the docker file:
```Dockerfile
# syntax=docker/dockerfile:1
FROM golang:1.17.7-bullseye

RUN apt-get update
RUN apt-get -y install python3
RUN apt-get -y install python3-setuptools
RUN apt-get -y install python3-pip

RUN pip install pre-commit==2.17.0
RUN pip install yamllint==1.26.3
RUN mkdir -p /opt/protolint
RUN curl -L https://github.com/yoheimuta/protolint/releases/download/v0.37.0/protolint_0.37.0_Linux_x86_64.tar.gz | tar -xz -C /opt/protolint
RUN ln -s /opt/protolint/protolint /usr/local/bin/protolint
```

The fail ci job is [pre-commit ci job](https://gitlab.com/DoneSpeak/container-image/-/jobs/2124031092).

another docker file is:
```Dockerfile
# syntax=docker/dockerfile:1
FROM python:3.8.12-buster

RUN pip install pre-commit==2.17.0
RUN pip install yamllint==1.26.3
RUN mkdir -p /opt/protolint
RUN curl -L https://github.com/yoheimuta/protolint/releases/download/v0.37.0/protolint_0.37.0_Linux_x86_64.tar.gz | tar -xz -C /opt/protolint
RUN ln -s /opt/protolint/protolint /usr/local/bin/protolint
```

**Comment by DoneSpeak** [comment]
Maybe I have figured out the reason. The parse requires a `syntax` key world in the protobuf file which I didn't provide. I think it should be a verify error like the following content instead of a code error which makes me very confused.

```
[tests/test.proto:4:3] Field "query" should avoid required for proto3
```

**Comment by yoheimuta** [comment]
@DoneSpeak 

> Maybe I have figured out the reason.

That's right.

>  I think it should be a verify error like the following content instead of a code error which makes me very confused.

I agree with you. The parser should make the error message clearer.

**Comment by CorrM** [comment]
Same here

![image](https://user-images.githubusercontent.com/5868825/182114425-1cf94472-1839-411f-bda2-7fbec49802b9.png)
![image](https://user-images.githubusercontent.com/5868825/182114508-dabf7259-8a5e-4fa7-8c89-2f2ce069d9de.png)

**Comment by ricsiLT** [comment]
Yours is different but related to protolint not being able to parse UTF-8-BOM files :D for now just save them as UTF-8 but it should be fixed - no reason this can't be handled.

**Comment by yoheimuta** [comment]
Thank you for clarifying the issue.

Handling UTF-8 BOM (U+FEFF) properly seems worth it.
Whenever I have time, I'm getting to it.

**Comment by gen2thomas** [comment]
Hi, first of all - many thanks for this tool!

I stumbled today across the same problem and had some trouble finding the cause. According the specification the BOM should be supported, so it makes sense to fix the error or at least improve the error message.

Thanks Thomas

https://protobuf.com/docs/language-spec#source-code-representation
